src/num/flpmath.c

```diff
@@ -3155,6 +3156,22 @@ void md_smax2(unsigned int D, const long dim[D], const long ostr[D], float* optr
 #else
 	make_3op_scalar(md_max2, D, dim, ostr, optr, istr, iptr, val);
 #endif
+#else
+	UNUSED(istr);
+	UNUSED(ostr);
+#ifdef USE_CUDA
+
+	if (cuda_ondevice(iptr)) {
+
+		assert(cuda_ondevice(optr));
+		gpu_ops.smax(md_calc_size(D, dim), optr, iptr, val);
+	}
+	else
+		cpu_ops.smax(md_calc_size(D, dim), optr, iptr, val);
+#else
+	cpu_ops.smax(md_calc_size(D, dim), optr, iptr, val);
```

Comment:
This does not work if the data is non-contiguous or has other non-trivial strides. This could be implemented exactly as 'md_smul' / 'md_zsmul' (but without the cuda special case - I am not even sure it is still needed in the other functions).